src/libraries/System.Reflection/tests/AssemblyTests.cs

```diff
@@ -154,10 +154,19 @@ public void GetEntryAssembly()
             Assert.NotNull(Assembly.GetEntryAssembly());
             string assembly = Assembly.GetEntryAssembly().ToString();
 
-            // The single file test runner is not xunit.console
-            string expectedAssembly = PlatformDetection.IsNativeAot ? "System.Reflection.Tests" : "xunit.console";
+            bool correct;
+            if (PlatformDetection.IsNativeAot)
+            {
+                // The single file test runner is not 'xunit.console'.
+                correct = assembly.IndexOf("System.Reflection.Tests", StringComparison.OrdinalIgnoreCase) != -1;
+            }
+            else
+            {
+                // Under Visual Studio, the runner is 'testhost', otherwise it is 'xunit.console'.
+                correct = assembly.IndexOf("xunit.console", StringComparison.OrdinalIgnoreCase) != -1 ||
```

Comment:
Please check if that actually works when running the test locally. I think the assembly name is xunit.console.dll only on .NETFramework and `Microsoft.DotNet.XUnitConsoleRunner.dll` on .NETCoreApp.

Reply:
What was the command that you used on the CLI? Was it `dotnet test ...`? Asking as we also support `dotnet build /t:Test ...` which uses the Microsoft.DotNet.XUnitConsoleRunner (.NETCoreApp) / xunit.console (.NETFramework) test host runner. That's also what is invoked on CI.

Reply:
Just the standard `build -test -s libs.tests` from the root.

I also ran just this with success:
```
.\dotnet.cmd build /t:Test .\src\libraries\System.Reflection\tests\System.Reflection.Tests.csproj /p:Configuration=Release
```

Reply:
Very weird. I was sure that the test runner would be Microsoft.DotNet.XUnitConsoleRunner.dll on .NETCoreApp. If I have some time I might debug this locally.

Reply:
OK, we only changed the PackageId but not the assembly name: https://github.com/dotnet/arcade/blob/ba1b5f7e9a6944cd26b24fb5634ab74ef2cbda98/src/Microsoft.DotNet.XUnitConsoleRunner/src/Microsoft.DotNet.XUnitConsoleRunner.csproj#L5.

That's why the assembly name is still xunit.console :) Sanity is back.